minio/api.py

```diff
@@ -1057,9 +1057,17 @@ def fget_object(self, bucket_name, object_name, file_path,
             )
 
             if progress:
-                # Set progress bar length and object name before upload
+                # Set progress bar length and object name before download
                 length = int(response.headers.get('content-length', 0))
-                progress.set_meta(object_name=object_name, total_length=length)
+                if not isinstance(progress, Thread):
+                    # Set progress bar length and object name before upload
+                    progress.set_meta(object_name=object_name, total_length=length)
+                elif hasattr(progress, "set_description") and hasattr(progress, "reset"):
```

Comment:
We could have generic progress bar support instead of this.
```py
class Progress:
    """A generic progress class."""
    __metaclass__ = ABCMeta

    @abstractmethod
    def set_meta(self, object_name, total_length):
        """Set progress metadata."""

    @abstractmethod
    def update(self, length):
        """Update progress."""
```
Here, check for `progress` is instance of `Progress`.

Reply:
`Progress` is only defined in examples for now.

I can define an abstract class instead, but I would follow this spec:
```
class Progress:
    """A generic progress class."""
    __metaclass__ = ABCMeta

    @abstractmethod
    def set_description(self, desc: str):
        """Set progress description."""

    @abstractmethod
    def reset(self, total: int):
        """Set progress description."""

    @abstractmethod
    def update(self, length):
        """Update progress."""
```

Reply:
This breaks backward compatibility which we don't want to do that. `tqdm` users can wrap `set_description()` and `reset()` into `set_meta()` of their own wrapper class.